### PR TITLE
[14.0][MIG] Migrate module account_invoice_ubl to v14

### PR DESCRIPTION
**Pull request by fshah-initos**
- [x] Depends on the PR #255

### CODE REVIEW

**Comment by PierrickBrun** [comment]
It depends on this PR https://github.com/OCA/edi/pull/255 if I understand correctly

File: account_invoice_ubl/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    "name": "Account Invoice UBL",
+    "version": "14.0.1.0.0",
+    "category": "Accounting & Finance",
+    "license": "AGPL-3",
+    "summary": "Generate UBL XML file for customer invoices/refunds",
+    "author": "Akretion,Onestein,Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/edi",
+    "depends": [
+        "account_e-invoice_generate",
```
Comment (CasVissers-360ERP) [inline account_invoice_ubl/__manifest__.py]: account_einvoice_generate module name has changed

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Dependency module name has changed

File: account_invoice_ubl/views/res_config_settings.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<!--
+  Copyright 2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
+<odoo>
+    <record id="view_account_config_settings" model="ir.ui.view">
+        <field name="model">res.config.settings</field>
+        <field
+            name="inherit_id"
+            ref="account_e-invoice_generate.view_account_config_settings"
```
Comment (CasVissers-360ERP) [inline account_invoice_ubl/views/res_config_settings.xml]: account_einvoice_generate

**Review by CasVissers-360ERP — APPROVED** [review]

**Review by rconjour — APPROVED** [review]

**Comment by fshah-initos** [comment]
@OCA/edi-maintainers can you please merge this PR ?

**Comment by CasVissers-360ERP** [comment]
@fshah-initos 
It seems like we are missing some data in the XML. Field "ref" is not added to the UBL while it should be added in the header like:
<cbc:BuyerReference>PO1234</cbc:BuyerReference>

Do you agree this should be part of the base module not an addition?

**Comment by fshah-initos** [comment]
I think we should keep it as it is as of now.

**Comment by CasVissers-360ERP** [comment]
I do not agree since it is not complete. But I will leave that up to the maintainers.

**Comment by CasVissers-360ERP** [comment]
BTW, this is the customer fix I made for now, you can see how simple it is:
```
    def _ubl_add_header(self, parent_node, ns, version="2.1"):
        res = super(AccountMove, self)._ubl_add_header(parent_node, ns, version="2.1")
        self.ensure_one()
        BuyerReference = etree.SubElement(parent_node, ns["cbc"] + "BuyerReference")
        BuyerReference.text = self.ref
        return res
```

**Comment by fshah-initos** [comment]
Hi @CasVissers-360ERP 
I've added the customer ref in the header. You can check

**Comment by CasVissers-360ERP** [comment]
@fshah-initos One minor improvement for invoices where ref is False:
buyer_reference.text = self.ref or ''

File: account_invoice_ubl/tests/test_ubl_generate.py [hunk 3]
```diff
@@ -3,15 +3,79 @@
...
+from odoo.tests.common import HttpCase
 
 from ..hooks import (
     remove_ubl_xml_format_in_pdf_invoice,
     set_xml_format_in_pdf_invoice_to_ubl,
 )
 
 
-class TestUblInvoice(TestAccountInvoice):
+class TestUblInvoice(HttpCase):
```
Comment (simahawk) [inline account_invoice_ubl/tests/test_ubl_generate.py:14]: ###

Comment (simahawk) [inline account_invoice_ubl/tests/test_ubl_generate.py:14]: Why HttpCase? There's no controller to test here. Please revert this change and use SavepointCase and setup the class like this to reduce overhead:

```py
@classmethod
def setUpClass(cls):
    super().setUpClass()
    cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```

**Review by simahawk — CHANGES_REQUESTED** [review]
please check the build

```
ERROR: TestUblInvoice.test_ubl_generate

Traceback (most recent call last):

  File "/home/travis/build/OCA/edi/setup/account_invoice_ubl/odoo/addons/account_invoice_ubl/tests/test_ubl_generate.py", line 87, in test_ubl_generate

    .render_qweb_pdf(invoice.ids)[0]

  File "/home/travis/odoo-14.0/addons/account/models/ir_actions_report.py", line 47, in render_qweb_pdf

    return super().render_qweb_pdf(res_ids=res_ids, data=data)

AttributeError: 'super' object has no attribute 'render_qweb_pdf'
```

**Comment by simahawk** [comment]
@fshah-initos ping :)

**Comment by simahawk** [comment]
@fshah-initos ping :)

**Review by TDu — COMMENTED** [review]
Migration looks good, there is one comment to take care of.
Also there is some new improvements on 13.0 to add here https://github.com/OCA/edi/commits/13.0/account_invoice_ubl
@fshah-initos

**Comment by simahawk** [comment]
@fshah-initos are you going to finish this?

**Comment by simahawk** [comment]
moving to https://github.com/OCA/edi/pull/482

**Pull request closed without merging**